src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1376,6 +1376,9 @@ private function getServiceCall($id, Reference $reference = null)
             return '$this';
         }
 
+        if ($this->container->hasDefinition($id) && !$this->container->getDefinition($id)->isPublic()) {
+            return "\$this->services[(isset(\$this->services['$id']) || \$this->{$this->generateMethodName($id)}()) && false ?: '$id']";
```

Comment:
What about a PHP version detection for generating? Or is this to much for such a simple case.